lib/kube/proxy/forwarder.go

```diff
@@ -613,6 +615,38 @@ func (f *Forwarder) setupContext(ctx auth.Context, req *http.Request, isRemoteUs
 	return authCtx, nil
 }
 
+// getKubeGroupsAndUsers returns the allowed kube groups/users names for a local kube cluster.
+func (f *Forwarder) getKubeGroupsAndUsers(
+	roles services.AccessChecker,
+	kubeClusterName string,
+	sessionTTL time.Duration) (groups, users []string, err error) {
+
+	kubeServices, err := f.cfg.CachingAuthClient.GetKubeServices(f.ctx)
+	if err != nil {
+		return nil, nil, trace.Wrap(err)
+	}
+
+	// Find requested kubernetes cluster name and get allowed kube users/groups names.
+	for _, s := range kubeServices {
+		for _, c := range s.GetKubernetesClusters() {
+			if c.Name != kubeClusterName {
+				continue
+			}
+
+			// Get list of allowed kube user/groups based on kubernetes service labels.
+			labels := types.CombineLabels(c.StaticLabels, c.DynamicLabels)
+			labelsMatcher := services.NewKubernetesClusterLabelMatcher(labels)
+			groups, users, err = roles.CheckKubeGroupsAndUsers(sessionTTL, false, labelsMatcher)
+			if err != nil {
+				return nil, nil, trace.Wrap(err)
+			}
+			return groups, users, nil
+		}
+	}
+	// kubeClusterName not found. Empty list of allowed kube users/groups is returned.
+	return
```

Comment:
Nit: I would explicitly `return []string{}, []string{}, nil`, naked returns make it hard to see what actually is being returned.